runtime/README.rst

```diff
@@ -5,7 +5,7 @@ Catalyst Quantum Runtime
 
 The Catalyst Runtime is a C++ QIR runtime that enables the execution of Catalyst-compiled
 quantum programs, and is currently backed by `PennyLane-Lightning <https://docs.pennylane.ai/projects/lightning/en/stable>`_
-state-vector simulators, and `Amazon Braket <https://amazon-braket-pennylane-plugin-python.readthedocs.io>`_
+state-vector simulators, and `Amazon Braket <https://amazon-braket-pennylane-plugin-python.readthedocs.io>`__
```

Comment:
Why two underscores? Most links only appear to have one

Reply:
These links were appearing twice :) I wanted to remove some of the sphinx warnings we had accumulated to make it easier to catch new warnings in the future

Reply:
I not sure I follow, what does the double underscore do? Should we use it going forward?

Reply:
Yep, best to always use double underscores. See here for a brief explanation: https://stackoverflow.com/questions/27420317/restructured-text-rst-http-links-underscore-vs-use

---

Basically, a single underscore will mean that Sphinx will try and treat the link as a named reference that you can re-use:

> With a single trailing underscore, the reference is named and the same target URI may be referred to again. With two trailing underscores, the reference and target are both anonymous, and the target cannot be referred to again. These are "one-off" hyperlinks.

This means that if you have the same link with the same text occurring twice on a page,

```rest
blah blah `Thing <https://thing.com>`_ and `Thing <https://thing.com/hello>`_
```

Sphinx will not be able to generate references, since the named reference `Thing` occurs more than once. It leads to a bunch of annoying warnings in the logs, _and_ could break links in certain situations.

If you use double underscores, Sphinx will not try and do this.

Reply:
Thanks for the explanation :) Would 
```
`Thing <https://thing.com>`_ and `Thing`_
```
then correctly produce the same link twice?

Reply:
Hmm I'm not sure. I think it's needs to be used like this:

<img width="832" alt="image" src="https://github.com/PennyLaneAI/catalyst/assets/2959003/e23a43a1-cf1a-4b1a-9d13-88768ed24103">

https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#embedded-uris-and-aliases

But I just default to using double underscores everywhere